validator_client/src/block_service.rs

```diff
@@ -852,6 +845,43 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
 
         Ok::<_, BlockError>(unsigned_block)
     }
+
+    /// translate `builder_proposals``, `builder_boost_factor`` and `prefer_builder_proposals`` to a
+    /// boost factor. if `builder_proposals` is set to false, set boost factor to 0 to indicate a
+    /// preference for local payloads. if `prefer_builder_proposals` is true, set boost factor to
+    /// u64::MAX to indicate a preference for builder payloads. if builder_boost_factor is a value other than none, return
+    /// its value as the boost factor. else return None to indicate no preference
+    /// between builder and local payloads
+    fn get_builder_boost_factor(&self, validator_pubkey: &PublicKeyBytes) -> Option<u64> {
+        let builder_proposals = self.validator_store.get_builder_proposals(validator_pubkey);
+
+        let builder_boost_factor = self
+            .validator_store
+            .get_builder_boost_factor(validator_pubkey);
+
+        let prefer_builder_proposals = self
+            .validator_store
+            .get_prefer_builder_proposals(validator_pubkey);
+
+        if !builder_proposals {
```

Comment:
The order here looks like if the `--builder-proposals` flag is forgotten, the `--prefer-builder-proposals` and `--builder-boost-factor` flags don't have any effect. Maybe we should use `requires` in `clap` to make sure `--builder-proposals` is provided with the other two. Alternatively, `--builder-proposals` could be implied by the presence of one of the other two flags.

Reply:
Yeah thats a good point. I think at the very least, `--builder-proposals` should be implied by the presence of one of the other two flags. Reason being is the validator management api and validator yml definition files wont trigger the `requires` in `clap`.